### PR TITLE
Add `*_like` array creation functions

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/dask/dask/issues/2637

Provides Dask Array equivalents of common NumPy `*_like` functions [`empty_like`]( https://docs.scipy.org/doc/numpy/reference/generated/numpy.empty_like.html ), [`ones_like`]( https://docs.scipy.org/doc/numpy/reference/generated/numpy.ones_like.html ), [`zeros_like`]( https://docs.scipy.org/doc/numpy/reference/generated/numpy.zeros_like.html ), and [`full_like`]( https://docs.scipy.org/doc/numpy/reference/generated/numpy.full_like.html ).

### CODE REVIEW

File: dask/array/creation.py [hunk 1]
```diff
@@ -12,8 +12,162 @@
 from ..base import tokenize
 from ..utils import ignoring
 from . import chunk
-from .core import Array, normalize_chunks, stack, concatenate
-from .wrap import empty
+from .core import Array, asarray, normalize_chunks, stack, concatenate
+from .wrap import empty, ones, zeros, full
+
+
+def empty_like(a, dtype=None, chunks=None):
```
Comment (jakirkham) [inline dask/array/creation.py]: These `*_like` functions in NumPy also take `order` and `subok` parameters. Though these didn't make much sense for Dask Array's IMHO as C-order arrays are used exclusively and am not aware of any subclass handling logic related to Dask Array's.

File: dask/array/creation.py [hunk 2]
```diff
@@ -12,8 +12,162 @@
...
+    zeros : Return a new array setting values to zero.
+
+    Notes
+    -----
+    This function does *not* initialize the returned array; to do that use
+    `zeros_like` or `ones_like` instead.  It may be marginally faster than
+    the functions that do set the array values.
+    """
+
+    a = asarray(a)
```
Comment (jakirkham) [inline dask/array/creation.py]: This means that we always have some kind of chunking. If not specified with `a` or the `chunks` argument, this ends up being the same as `a.shape`. This seems like reasonable behavior though we could raise for unspecified chunks instead.

**Comment by jakirkham** [comment]
Personally am happy with this, but have noted two pieces that could be dealt with differently. Would be interested to hear feedback from others on these points or other aspects of interest.

**Comment by shoyer** [comment]
This approach looks rights to me.

File: dask/array/creation.py (hunk 1 above)
Comment (mrocklin) [reply dask/array/creation.py]: Agreed

File: dask/array/creation.py (hunk 2 above)
Comment (mrocklin) [reply dask/array/creation.py]: Using the input's chunks if not specified seems good to me

**Comment by mrocklin** [comment]
+1

Merging tomorrow if there are no further comments

**Comment by mrocklin** [comment]
These are in.  Thanks @jakirkham !

**Comment by jakirkham** [comment]
Thanks @mrocklin and @shoyer.

**Pull request merged**